bugbug/models/accessibility.py

```diff
@@ -0,0 +1,160 @@
+from sklearn.pipeline import Pipeline
+
+from bugbug import bug_features, bugzilla, feature_cleanup, utils
+from bugbug.model import BugModel
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class AccessibilityModel(BugModel):
+    def __init__(self, lemmatization=False):
+        BugModel.__init__(self, lemmatization)
+
+        self.calculate_importance = False
+
+        feature_extractors = [
+            bug_features.HasSTR(),
+            bug_features.Keywords({"access"}),
+            bug_features.HasAttachment(),
+            bug_features.Product(),
+            bug_features.FiledVia(),
+            bug_features.HasImageAttachment(),
+        ]
+
+        cleanup_functions = [
+            feature_cleanup.fileref(),
+            feature_cleanup.url(),
+            feature_cleanup.synonyms(),
+        ]
+
+        self.extraction_pipeline = Pipeline(
+            [
+                (
+                    "bug_extractor",
+                    bug_features.BugExtractor(
+                        feature_extractors,
+                        cleanup_functions,
+                        rollback=True,
+                    ),
+                ),
+            ]
+        )
+
+        self.clf = ImblearnPipeline(
+            [
+                (
+                    "union",
+                    ColumnTransformer(
+                        [
+                            ("data", DictVectorizer(), "data"),
+                            ("title", self.text_vectorizer(min_df=0.0001), "title"),
+                            (
+                                "first_comment",
+                                self.text_vectorizer(min_df=0.0001),
+                                "first_comment",
+                            ),
+                        ]
+                    ),
+                ),
+                ("sampler", SMOTE(random_state=0)),
+                (
+                    "estimator",
+                    xgboost.XGBClassifier(
+                        n_jobs=utils.get_physical_cpu_count(),
+                        scale_pos_weight=9,
+                        subsample=0.5,
+                    ),
+                ),
+            ]
+        )
+
+    @staticmethod
+    def __is_accessibility_bug(bug):
+        """Check if a bug is an accessibility bug."""
+        return bug["cf_accessibility_severity"] != "---" or "access" in bug["keywords"]
+
+    @staticmethod
+    def __download_older_access_bugs(years: int, months: int) -> None:
+        """Retrieve accessibility related bugs from a specific time range.
+
+        By including older accessibility bugs, this function extends the dataset used
+        for model training compared to the default, which only considers bugs from 2 years
+        and 6 months ago. This extension in the time frame aims to improve the performance
+        of the model by providing a more comprehensive set of historical data.
```

Comment:
It is outdated, and we could make it shorter or move it to where we use the function.